pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
> @deads2k I have only very foggy recollection of this code. Was this a FYI or are you asking me for something?

They're talking about removing the kube serialized type as some kind of followup.  I'd like to keep our specific serialized type, but figured you'd have an opinion.

@lavalamp  Should we keep our serialization type to control the format and simply write tests to track drift between our types and upstream types?  (we do this in openshift and it works reasonably well)